internal/resource/types.go

```diff
@@ -39,4 +39,5 @@ type Device interface {
 	GetTotalMemoryMB() (uint64, error)
 	GetDeviceHandleFromMigDeviceHandle() (Device, error)
 	GetCudaComputeCapability() (int, int, error)
+	GetClass() (string, error)
```

Comment:
Should this be:
```suggestion
	GetPCIClass() (int, error)
```
Alternatively we can just have this be `GetMode() (string, error)`

Reply:
Aren't all these devices PCIE. I feel  `GetClass() (int, error)` makes more sense.

Reply:
All the devices are PCIE so I feel GetClass makes more sense.

Reply:
The point is that from the point of view of the caller Class is very generic. We have NVML devices, CUDA Devices -- which include Tegra iGPUs, and SYSFS devices. Adding PCIE to the function name clarifies that we're talking about a class.

Reply:
Renamed the method